File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function check_subp(dic) {
    for (let val of Object.values(dic)) {
        if (val > 0) {
            return false
        }
    }
    return true
}

function solution(want, number, discount) {
    var answer = 0;
    let cust_product = {}
    for (let i=0; i < want.length; i++) {
        cust_product[want[i]] = number[i]
    }
    for (let i = 0; i < 10; i++) {
        if ((discount[i] in cust_product)) {
            cust_product[discount[i]] -= 1
        }
    }
    answer += check_subp(cust_product)
    for (let i=0; i < discount.length-10; i++) {
        if (discount[i] in cust_product) {
            cust_product[discount[i]] += 1
        }
        if (discount[i+10] in cust_product) {
            cust_product[discount[i+10]] -= 1
        } 
        answer += check_subp(cust_product)
    }
    return answer;
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def check_subp(dic):
    for val in dic.values():
        if val > 0:
            return False
    return True

def solution(want, number, discount):
    answer = 0
    cust_product = dict()
    for i in range(len(want)):
        cust_product[want[i]] = number[i]
    for i in range(0,10):
        if discount[i] in cust_product.keys():
            cust_product[discount[i]] -= 1
    answer += check_subp(cust_product)
    for i in range(0, len(discount)-10):
        if discount[i] in cust_product.keys():
            cust_product[discount[i]] += 1
        if discount[i+10] in cust_product.keys():
            cust_product[discount[i+10]] -= 1
        answer += check_subp(cust_product)
    return answer